refactor(formatter, codegen): use `Comment::newlines` to checking if there is a newline around

Dunqing opened the pull request:
`Comment::preceded_by_newline` is the same as `is_own_line_comment`
`Comment::followed_by_newline` is the same as `is_end_of_line_comment`

This PR removes `is_own_line_comment` and `is_end_of_line_comment` methods, and changes usages to use `Comment::preceded_by_newline` and `Comment::followed_by_newline` to eliminate extra line break checking.

leaysgur commented:
### Merge activity

* **Dec 4, 7:25 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 4, 7:25 AM UTC**: `leaysgur` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Dunqing committed:
refactor(formatter, codegen): use `Comment::newlines` to checking if there is a newline around (#16482)

`Comment::preceded_by_newline` is the same as `is_own_line_comment`
`Comment::followed_by_newline` is the same as `is_end_of_line_comment`

This PR removes `is_own_line_comment` and `is_end_of_line_comment` methods, and changes usages to use `Comment::preceded_by_newline` and `Comment::followed_by_newline` to eliminate extra line break checking.
